### PR TITLE
Added save ability for Web (Fix #143 & Fix #145)

### LINKED ISSUE #143: Save not working
# Overview
I tried saving, but I got this error:
image

# Device
* Web version
* Windows 11

### LINKED ISSUE #145: Can't open file on the web version
The file system API hasn't been implemented for the open feature.

### PR DESCRIPTION
**Pull request by osmium-dev**
# Overview
I implemented the save system for the web version using the experimental [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API).

### Verel Preview
https://rosepine-git-fork-osmium-dev-main-thien-do.vercel.app/
### Video
https://user-images.githubusercontent.com/99157490/186921166-153f3c73-a9f5-40cc-a784-414bebc12cf9.mp4

## How to use.
This feature is experimental so you will have to enable it. Also, the API requires SSL, so you must either add SSL to your localhost or use something like [localtunnel](https://theboroer.github.io/localtunnel-www/) or [ngrok](https://ngrok.com/).
In other words:
- Enable the File System Access Flag (chrome://flags/#file-system-access-api)
- Add SSL to localhost ([tuto](https://www.section.io/engineering-education/how-to-get-ssl-https-for-localhost/))
or
- Use [localtunnel](https://theboroer.github.io/localtunnel-www/) or [ngrok](https://ngrok.com/).

### CODE REVIEW

**Comment by thien-do** [comment]
Thanks! I'm reviewing it now!

**Review by thien-do — APPROVED** [review]
Thanks!
